components/reddit/sources/new-link-on-a-subreddit/new-link-on-a-subreddit.js

```diff
@@ -0,0 +1,81 @@
+const reddit = require("../../reddit.app.js");
+
+module.exports = {
+  key: "new-link-on-a-subreddit",
+  name: "New Link on a subreddit",
+  description: "Emits an event each time a new link is added to a subreddit",
+  version: "0.0.1",
+  dedupe: "unique",
+  props: {
+    reddit,
+    subreddit: { propDefinition: [reddit, "subreddit"] },  
+    timer: {
+      label: "Polling schedule",
+      description: "Pipedream polls Reddit for new links on this schedule.",
+      type: "$.interface.timer",
+      default: {
+        intervalSeconds: 60 * 10, // by default, run every 10 minutes.
+      },
+    },
+    db: "$.service.db",
+  },
+  hooks: {
+    async deploy() {
+      // Emits 10 sample events on the first run during deploy.
+      let before = null;
+
+			try{
+	      var reddit_things = await this.reddit.getNewSubredditLinks(
+	        null,
+	        this.subreddit,        
+	        10
+	      );
+    	}catch (err) {
+			  if (err.response.status) {
+			    throw new Error(`We encountered a 404 error trying to fetch links for ${this.subreddit}. Please check the subreddit name and try again`);
+			  }
+			  throw err;
+			}
+
+      const links_pulled = this.reddit.wereLinksPulled(reddit_things);
+      if (links_pulled) {
+        before = reddit_things.data.children[0].data.name;
+    	const ordered_reddit_things = reddit_things.data.children.reverse();
+        ordered_reddit_things.forEach((reddit_link) => {
+          this.emitRedditEvent(reddit_link);
+        });
+      }
+
+      this.db.set("before", before);
+    },
+  },
+  methods: {
+    emitRedditEvent(reddit_event) {
+      var { name: id, title: summary } = reddit_event.data;
+      summary = new Date(reddit_event.data.created);
```

Comment:
Did you mean to change the summary here? Or were you intending to set the `ts` parameter on the emit?

Reply:
Oh sorry, no. It was code for my debugging. 

I am leaving the summary without changes. However,  I raised a timestamp question in comment 5 days ago. I am copying here for easiness, do you have advice on the timestamp question?

> Ok, I decided to implement pagination in this method. However, I don't see how the timestamp is relevant here. The reddit objects returned by the API endpoint for this source don't have a field related to a last update/modify date (I am thinking in the strategy used in the sample mercury source with code like
> 
>  if (updatedAt.getTime() > updatedAfter.getTime()) {
>           this.$emit(contact, this.generateMeta(contact, updatedAt));
>         } else {
>           // don't need to continue if we've gotten to contacts already evaluated
>           done = true;
>         }
> As per reddit API documentation, for pagination only after or before are given (optionally use a count to keep truck of the total number of records having been pulled)
> 
> To page through a listing, start by fetching the first page without specifying values for after and count. The response will contain an after value which you can pass in the next request. It is a good idea, but not required, to send an updated value for count which should be the number of items already fetched.
> 
> Could you elaborate on the use of the timestamp? I can think it can be included in the metadata for the event to indicate when the event is being emitted

Reply:
as discussed on Slack, timestamps in the emitted event object are optional but recommended.

```
Could you elaborate on the use of the timestamp? I can think it can be included in the metadata for the event to indicate when the event is being emitted

yes, the timestamp is optional but recommended. It corresponds to the timestamp that's tied to the event, and in this case, I agree createdAt sounds like the best choice.
ts accepts an integer corresponding to the epoch millisecond value for the timestamp, e.g. 1614204756487
```